Content.Shared/Ghost/GhostComponent.cs

```diff
@@ -0,0 +1,117 @@
+        };
+
+        public InstantAction ToggleGhostsAction = new()
+        {
+            Icon = new SpriteSpecifier.Rsi(new ("Mobs/Ghosts/ghost_human.rsi"), "icon"),
+            DisplayName = "ghost-gui-toggle-ghost-visibility-name",
+            Description = "ghost-gui-toggle-ghost-visibility-desc",
+            ClientExclusive = true,
+            CheckCanInteract = false,
+            Event = new ToggleGhostsActionEvent(),
+        };
+
+        public TimeSpan TimeOfDeath { get; set; } = TimeSpan.Zero;
+
+        [DataField("booRadius")]
+        public float BooRadius = 3;
+
+        [DataField("booMaxTargets")]
+        public int BooMaxTargets = 3;
+
+        [DataField("action")]
+        public InstantAction Action = new()
+        {
+            UseDelay = TimeSpan.FromSeconds(120),
+            Icon = new SpriteSpecifier.Texture(new ("Interface/Actions/scream.png")),
+            DisplayName = "action-name-boo",
+            Description = "action-description-boo",
+            CheckCanInteract = false,
+            Event = new BooActionEvent(),
+        };
+
+        // TODO: instead of this funny stuff just give it access and update in system dirtying when needed
+        [ViewVariables(VVAccess.ReadWrite)]
+        public bool CanGhostInteract
+        {
+            get => _canGhostInteract;
+            set
+            {
+                if (_canGhostInteract == value) return;
+                _canGhostInteract = value;
+                Dirty();
+            }
+        }
+
+        [DataField("canInteract"), AutoNetworkedField]
+        private bool _canGhostInteract;
+
+        /// <summary>
+        ///     Changed by <see cref="SharedGhostSystem.SetCanReturnToBody"/>
+        /// </summary>
+        // TODO MIRROR change this to use friend classes when thats merged
+        [ViewVariables(VVAccess.ReadWrite)]
+        public bool CanReturnToBody
+        {
+            get => _canReturnToBody;
+            set
+            {
+                if (_canReturnToBody == value) return;
+                _canReturnToBody = value;
+                Dirty();
+            }
+        }
+
+        /// <summary>
+        /// Ghost color
+        /// </summary>
+        /// <remarks>Used to allow admins to change ghost colors. Should be removed if the capability to edit existing sprite colors is ever added back.</remarks>
+        [DataField("color"), AutoNetworkedField]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public Color color = Color.White;
+
+        [DataField("canReturnToBody"), AutoNetworkedField]
+        private bool _canReturnToBody;
+
+        public sealed partial class ToggleLightingActionEvent : InstantActionEvent { };
+
+        public sealed partial class ToggleFoVActionEvent : InstantActionEvent { };
+
+        public sealed partial class ToggleGhostsActionEvent : InstantActionEvent { };
+
+        public sealed partial class BooActionEvent : InstantActionEvent { }
```

Comment:
shouldnt these be outside the component class

Reply:
legacy but sure.